webapp/public/js/domjudge.js

```diff
@@ -429,9 +429,12 @@ function processAjaxResponse(jqXHR, data) {
         var newCurrentContest = parseInt(jqXHR.getResponseHeader('X-Current-Contest'));
         var dataCurrentContest = $('[data-current-contest]').data('current-contest');
         var currentContest = parseInt(dataCurrentContest);
+        var refreshStop = $('[data-ajax-refresh-stop]');
 
-        // If the contest ID changed from another tab, reload or whole page
-        if (dataCurrentContest !== undefined && newCurrentContest !== currentContest) {
+        // Reload the whole page if
+        // - we were signaled to stop refreshing, or
+        // - if the contest ID changed from another tab.
+        if ("0" in refreshStop || (dataCurrentContest !== undefined && newCurrentContest !== currentContest)) {
```

Comment:
We set refreshStop to 1, so why are we checking for refreshStop? Also shouldn't the line above be:
```
var refreshStop = $('[data-ajax-refresh-stop]').data('ajax-refresh-stop');
```
?

Reply:
I was basically testing whether this signal was there at all (had a key `0`), but yeah this is nicer. Fixed.